lib/src/rules/unnecessary_parenthesis.dart

```diff
@@ -78,6 +78,17 @@ class _Visitor extends SimpleAstVisitor<void> {
       // whitespace after its first token.
       if (parent is PrefixExpression &&
           _expressionStartsWithWhitespace(node.expression)) return;
+
+      // Another case of the above exception, something like
+      // `!(const [7]).contains(5);`, where the _parent's_ parent is the
+      // PrefixExpression.
+      if (parent is MethodInvocation) {
+        Expression target = parent.target;
+        if (parent.parent is PrefixExpression &&
+            target is ParenthesizedExpression &&
```

Comment:
While I think that this is technically correct, I think we want to use `target == node` (rather than `target is ParenthesizedExpression`) here. I think that will make it clear that the interesting condition is that the node we're looking at is the target of the invocation. (I think it's technically correct because I can't think of any other child of a `MethodInvocation` that _could_ be a parenthesized expression, but if that ever changes then this would be broken.)

Reply:
Ah, good catch! Done.